backend/modules/professor/routers/professor_grading.py

```
@@ -0,0 +1,403 @@
+        ).filter(
+            StudentSimulationInstance.id == instance_id
+        ).first()
+        
+        if not instance:
+            raise HTTPException(status_code=404, detail="Simulation instance not found")
+        
+        # Verify the professor has access to this instance
+        if instance.cohort_assignment:
+            cohort = instance.cohort_assignment.cohort
+            if cohort and cohort.created_by != current_user.id:
+                raise HTTPException(status_code=403, detail="Access denied")
```

Ruff (0.14.8)

Abstract `raise` to an inner function

(TRY301)